media/js/products/vpn/resource-center-article.es6.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// Lazyload images
+Mozilla.LazyLoad.init();
+
+//event handlers for voting widget on article pages
+let hasVoted = false;
+const upvoteBtn = document.querySelector('.vpn-c-vote-btn.up');
+const downvoteBtn = document.querySelector('.vpn-c-vote-btn.down');
+
+function trackVoteInteraction(url, vote) {
+    window.dataLayer.push({
+        event: 'vpn-article-vote',
+        label: url,
+        value: vote
+    });
+}
+
+function handleVote({ target }) {
+    const isUpvote = upvoteBtn.contains(target);
+    if (
+        ((isUpvote && upvoteBtn.classList.contains('active')) ||
```

Comment:
This is a tough condition to follow. Maybe an easier check would be if either button has `aria-pressed=true`?

Also, do we want to allow people to change their votes all the time? Or did we only want them to vote once?
Right now, because we set the active class below, if a person has already voted down and decides to vote up, they will be allowed to vote again (it is an upvote and hasVoted is true, but the upvote button does not have active class so the condition is false)

Reply:
I'll double check with others, but I think that we wanted to allow the user to change their vote between the two but not to be able to continue spamming the button in GA